base_guicontrol.cpp

```diff
@@ -1,45 +1,53 @@
-#include <wx/wxprec.h>
-
-#ifndef WX_PRECOMP
-#  include <wx/wx.h>
-#endif
-
-#include "base.h"
-#include <string>
-
-//////////////////////////////////
-// METHODS FOR MANIPULATING GUI //
-//////////////////////////////////
-
-void MainFrame::clearOutput(){
-  outputText->SetValue("");
-}
-
-void MainFrame::printToOutput(std::string& text){
-  outputText->SetValue(text);
-}
-
-void MainFrame::appendOutput(std::string& text){
-  outputText->SetValue(outputText->GetValue() + text);
-}
-
-std::string MainFrame::getAtomFilepath(){
-  return filepathText->GetValue().ToStdString();
-}
-
-std::string MainFrame::getRadiusFilepath(){
-  return radiuspathText->GetValue().ToStdString();
-}
-
-double MainFrame::getGridsize(){
-  return std::stod(gridsizeInputText->GetValue().ToStdString());
-}
-
-int MainFrame::getDepth(){
-  return depthInput->GetValue();
-}
-
-//* get from user input
-double MainFrame::getProbeRadius(){
-  return r_probe;
-}
+#include <wx/wxprec.h>
+
+#ifndef WX_PRECOMP
+#  include <wx/wx.h>
+#endif
+
+#include "base.h"
+#include <string>
+
+//////////////////////////////////
+// METHODS FOR MANIPULATING GUI //
+//////////////////////////////////
+
+void MainFrame::clearOutput(){
+  outputText->SetValue("");
+}
+
+void MainFrame::printToOutput(std::string& text){
+  outputText->SetValue(text);
+}
+
+void MainFrame::printToOutputUnicode(std::wstring& text){
+  outputText->SetValue(text);
+}

```

Comment:
It is simpler here to overload the function, rather than make a whole new stand-alone function. Overloading a function means using the same function name with a different (set of) input arguments.

Also, if you have two functions that essentially do the same thing it is typically better for upkeep to define one function in terms of the other. In this case, I think the std::string function should call the std::wstring function, assuming string can be cast to wstring.